app/assets/v2/js/cart-ethereum-zksync.js

```diff
@@ -0,0 +1,273 @@
+    donationInputs: { type: Array, required: true }, // donationInputs computed property from cart.js
+    network: { type: String, required: true } // web3 network to use
+  },
+
+  data: function() {
+    return {
+      zksync: {
+        checkoutManager: undefined, // zkSync API CheckoutManager class
+        feeTokenSymbol: undefined, // token symbol to pay zksync fee with, e.g. 'DAI'
+        showChangeToken: false, // true to show dropdown to let user select feeTokenSymbol
+        showModal: false, // true to show modal to user, false to hide
+        checkoutStatus: 'not-started' // options are 'not-started', 'pending'. When complete we redirect the user
+      },
+
+      cart: {
+        tokenList: [], // array of tokens in the cart
+        unsupportedTokens: [] // tokens in cart which are not supported by zkSync
+      },
+
+      user: {
+        address: undefined, // connected web3 wallet address
+        zkSyncState: undefined, // contains account balances of zkSync wallet
+        hasEnoughBalance: true
+      }
+    };
+  },
+
+  mounted() {
+    // If user started checking out with zkSync, warn them before closing or reloading page.
+    // Note that beforeunload may sometimes be ignored by browsers, e.g. if users have not
+    // interacted with the page
+    window.addEventListener('beforeunload', (e) => {
+      if (this.zksync.checkoutStatus === 'pending') {
+        // This shows a generic message staying "Leave site? Changes you made may not be saved". This
+        // cannot be changed, as "the ability to do this was removed in Chrome 51. It is widely
+        // considered a security issue, and most vendors have removed support."
+        // source: https://stackoverflow.com/questions/40570164/how-to-customize-the-message-changes-you-made-may-not-be-saved-for-window-onb
+        e.preventDefault();
+        e.returnValue = '';
+      }
+    });
+  },
+
+  computed: {
+    /**
+     * @dev List of tokens supported by zkSync + Gitcoin. To add a token to this list:
+     *   1. Make sure the token is supported by zkSync
+     *        Mainnet list: https://api.zksync.io/api/v0.1/tokens
+     *        Rinkeby list: https://rinkeby-api.zksync.io/api/v0.1/tokens
+     *   2. Add the token symbol to the appropriate list below
+     * @dev We hardcode the list here instead of fetching from the API to improve performance and
+     * reduce problems that arise if zkSync's server is slow
+     */
+    supportedTokens() {
+      const mainnetTokens = [ 'ETH', 'DAI', 'USDC', 'TUSD', 'USDT', 'SUSD', 'BUSD', 'LEND', 'BAT', 'KNC', 'LINK', 'MANA', 'MKR', 'REP', 'SNX', 'WBTC', 'ZRX', 'MLTT', 'LRC', 'HEX', 'PAN', 'SNT', 'YFI', 'UNI', 'STORJ', 'TBTC', 'EURS', 'GUSD', 'RENBTC', 'RNDR', 'DARK', 'CEL', 'AUSDC', 'CVP', 'BZRX', 'REN' ];
+      const rinkebyTokens = [ 'ETH', 'USDT', 'USDC', 'LINK', 'TUSD', 'HT', 'OMG', 'TRB', 'ZRX', 'BAT', 'REP', 'STORJ', 'NEXO', 'MCO', 'KNC', 'LAMB', 'GNT', 'MLTT', 'XEM', 'DAI', 'PHNX' ];
```

Comment:
I wonder if it might be worth stuffing these in a JSON blob in the database. The reason I say this is, as it stands, we need to do a PR + redeploy to support a new token. With something in the DB, it could be edited in the admin w/o updating code. What do you think?

Reply:
That's a good point, and you're right it's probably the better approach. @thelostone-mc I know this is how regular tokes are configured, so what's involved with doing that for these lists as well? 

On the other hand, considering over 99.9% of donations are made in tokens already on this list, it may not be worth the effort right now

Reply:
hmm having a new token table just for zksync seems like an overkill given that it won't change much 🤔
if we support other L2 (not sure about that) -> would it mean having a new table for every L2 ?

Reply:
You know, I think you're right @thelostone-mc, especially per @mds1's point that 99% of donations come from just a few tokens. I withdraw my suggestion!

Reply:
i'd lean towards keeping it as a list in code for now and switch it over later is needed

Reply:
Sounds good, I'll mark this as resolved then 👍